ksa_compliance/standard_doctypes/sales_invoice.py

```diff
@@ -107,4 +107,4 @@ def validate_sales_invoice(self, method) -> None:
             valid = False
 
     if not valid:
-        raise frappe.ValidationError()
+        raise frappe.ValidationError(_(f"Please set a default sales taxes and charges template for {self.company}."))
```

Comment:
You need to raise the actual error that happened depending on the conditions above. You can either create an error list and append errors to it when you msgprint, or you can call `frappe.get_message_log` which returns a list of messages previously passed to msgprint (each is a dict with the message in the  `message` key)